serg/mtr-bash-completion.sh

```diff
@@ -1,30 +1,30 @@
 _mtr_complete_testnames ()
 {
-  dir=$1
-  [ -d $dir/t ] && dir=$dir/t
-  testnames=$( cd $dir && echo *.test | sed -e 's/\.test\>//g' )
+  dir="$1"
+  [ -d "$dir"/t ] && dir="$dir"/t
+  testnames=$( cd "$dir" && echo *.test | sed -e 's/\.test\>//g' )
 }
 _mtr_all_suites ()
 {
-  suites=$(find suite ../{storage,plugin}/*/mysql-test -type d -exec find '{}' -maxdepth 1  -name '*.test' -print -quit \; | sed -E 's@/(t/)?[^/]+$@@; s@^(suite|.*/mysql-test)/@@'|sort -u)
+  suites=$(find "$sourcetestdir"/suite "$sourcetestdir"/../{storage,plugin}/*/mysql-test -type d -exec find '{}' -maxdepth 1  -name '*.test' -print -quit \; | sed -E "s@^$sourcetestdir/(suite|\.\./storage|\.\./plugin)/@@"'; s@/mysql-test@@; s@(/t)?/[^/]+\.test$@@'|sort -u)
 }
 _mtr_complete()
 {
-  [ -x ./mtr ] || return
-  [ -d main ] && main=main || main=.
+  sourcetestdir=$(sed -n -e "/^chdir/s/[^']*'\(.*\)');$/\1/p" < "$1")
```

Comment:
that works only for out-of-source `mtr`, right? what about "normal" in-source `mysql-test-run.pl` ?